File: En Python - Interpretado/PrimeraVocal.py
# -*- coding: utf-8 -*-

check = True
leng = 1

palabra = input("Ingrese la palabra a evaluar: ")

for p in palabra:
    if p.lower() == 'a' or p.lower() == 'á':
        print("La primera vocal es: {0}".format(p))
        break
    elif p.lower() == 'e' or p.lower() == 'é':
        print("La primera vocal es: {0}".format(p))
        break
    elif p.lower() == 'i' or p.lower() == 'í':
        print("La primera vocal es: {0}".format(p))
        break
    elif p.lower() == 'o' or p.lower() == 'ó':
        print("La primera vocal es: {0}".format(p))
        break
    elif p.lower() == 'u' or p.lower() == 'ú':
        print("La primera vocal es: {0}".format(p))
        break
    elif check:
        if leng == len(palabra):
             print("Lo sentimos, no hay vocales!!")
        leng += 1

File: En C++ - Compilado/PrimeraVocal.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    string palabra;
    cout << "Ingrese la palabra a evaluar: ";
    getline(cin >> ws, palabra);

    for(int id = 0; id < palabra.length(); id++) {
        if(palabra[id] == 'a' || palabra[id] == 'A') {
            cout << "La primera vocal es: " << palabra[id] << endl;
            break;
        } else if(palabra[id] == 'e' || palabra[id] == 'E') {
            cout << "La primera vocal es: " << palabra[id] << endl;
            break;
        } else if(palabra[id] == 'i' || palabra[id] == 'I') {
            cout << "La primera vocal es: " << palabra[id] << endl;
            break;
        } else if(palabra[id] == 'o' || palabra[id] == 'O') {
            cout << "La primera vocal es: " << palabra[id] << endl;
            break;
        } else if(palabra[id] == 'u' || palabra[id] == 'U') {
            cout << "La primera vocal es: " << palabra[id] << endl;
            break;
        } else if(palabra.length() == id) {
            cout << "Lo sentimos, no hay vocales!" << endl;
        }
    }
    return 0;
}
